client/src/components/HostMenu.jsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import io from 'socket.io-client';
+import Dashboard from './Dashboard.jsx';
+
+const url = 'http://localhost:8080';
+const socket = io(url);
+
+const HostMenu = ({ guests, menu, setTopLevelState }) => {
+  const [currentName, setCurrentName] = useState('');
+
+  useEffect(() => {
+    socket.on('orderSubmitted', payload => {
+      console.log('Orders: ', payload);
+      const guestArray = [...guests, payload];
+      setTopLevelState('guests', guestArray);
+    });
+  });
+
+  const onChange = (event) => {
+    setCurrentName(event.target.value);
+    onCloseClick();
+    event.preventDefault();
+  };
+
+  const addItem = (item) => {
+    const guestArray = [...guests];
+    for (let i = 0; i < guestArray.length; i++) {
+      if (guestArray[i].guestName === currentName) {
+        if (guestArray[i].order.find(menuItem => menuItem.name === item.name)) {
+          alert('Item already added');
```

Comment:
I'm wondering if we really need to prevent the same item being added multiple times. I think it's entirely possible someone might get more than 1 of an item during a meal (e.g. more than 1 drink, an extra dessert, more coffee, etc).

Reply:
Good catch. We made sure to include the quantity as a property in the guest experience flow and then totally forgot to implement it in the HostMenu component. I'll change this to use the same logic that exists in the GuestMenu component.